rotkehlchen/types.py

```diff
@@ -1050,6 +1056,54 @@ def deserialize(cls: type['LocationAssetMappingUpdateEntry'], data: dict[str, An
         except KeyError as e:
             raise DeserializationError(f'Missing key {e!s}') from e
 
+    def serialize_for_db(self) -> tuple[Any, ...]:
+        values = super().serialize_for_db()
+        return self.asset.serialize(), *values
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=True)
+class CounterpartyAssetMappingDeleteEntry:
+    """A counterparty asset mapping delete entry"""
+    counterparty: str
+    counterparty_symbol: str
+
+    @classmethod
+    def deserialize(cls: type['CounterpartyAssetMappingDeleteEntry'], data: dict[str, Any]) -> 'CounterpartyAssetMappingDeleteEntry':  # noqa: E501
+        try:
+            return cls(
+                counterparty=data['counterparty'],
+                counterparty_symbol=data['counterparty_symbol'],
+            )
+        except KeyError as e:
+            raise DeserializationError(f'Missing key {e!s}') from e
+
+    def serialize_for_db(self) -> tuple[str, ...]:
+        return self.counterparty_symbol, self.counterparty
+
+    def __str__(self) -> str:
+        return f'{self.counterparty_symbol} for {self.counterparty}'
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=True)
+class CounterpartyAssetMappingUpdateEntry(CounterpartyAssetMappingDeleteEntry):
+    """A counterparty asset mapping update entry"""
+    asset: 'Asset'
+
+    @classmethod
+    def deserialize(cls: type['CounterpartyAssetMappingUpdateEntry'], data: dict[str, Any]) -> 'CounterpartyAssetMappingUpdateEntry':  # noqa: E501
+        try:
+            return cls(
+                asset=data['asset'],
+                counterparty=data['counterparty'],
+                counterparty_symbol=data['counterparty_symbol'],
+            )
+        except KeyError as e:
+            raise DeserializationError(f'Missing key {e!s}') from e
+
+    def serialize_for_db(self) -> tuple[str, ...]:
+        values = super().serialize_for_db()
+        return self.asset.serialize(), *values
+
```

Comment:
can be a single line?

```suggestion
        return self.asset.serialize(), *super().serialize_for_db()

```